### PR TITLE
Warn on access token reveal

### PR DESCRIPTION
**Pull request by resynth1943**
Fixes https://github.com/vector-im/element-web/issues/15367

image

Signed-off-by: Resynth <resynth1943@tutanota.com>

### CODE REVIEW

**Comment by resynth1943** [comment]
This *does* include a new `_t('...')` string. Since it's the weekend and nobody is manning the channels, I'm not entirely sure if I've added this correctly ...?

File: src/components/views/dialogs/AccessTokenDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+/*
+Copyright 2017 Vector Creations Ltd
```
Comment (t3chguy) [inline src/components/views/dialogs/AccessTokenDialog.tsx:2]: You should specify your copyright here

File: src/components/views/dialogs/AccessTokenDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+*/
+
+import React from 'react';
+import { _t } from '../../../languageHandler';
+import QuestionDialog from './QuestionDialog';
+
+type IProps = Exclude<
+    React.ComponentProps<QuestionDialog>,
+    "title" | "danger" | "description"
+    >;
```
Comment (t3chguy) [inline src/components/views/dialogs/AccessTokenDialog.tsx:24]: this line-breaking looks a bit strange

**Review by t3chguy — APPROVED** [review]

**Comment by t3chguy** [comment]
Looks good from a code perspective otherwise, a Screenshot will likely help with the Design review

**Comment by resynth1943** [comment]
Oops on the commit name.

File: src/components/views/dialogs/AccessTokenDialog.tsx (hunk 2 above)
Comment (resynth1943) [reply src/components/views/dialogs/AccessTokenDialog.tsx]: You alright with https://github.com/matrix-org/matrix-react-sdk/pull/5355/commits/a3212c0477ec0fc9166a56a49a9579bae4712d5f?

File: src/components/views/dialogs/AccessTokenDialog.tsx (hunk 1 above)
Comment (resynth1943) [reply src/components/views/dialogs/AccessTokenDialog.tsx:2]: https://github.com/matrix-org/matrix-react-sdk/pull/5355/commits/a3212c0477ec0fc9166a56a49a9579bae4712d5f thanks :D

**Comment by resynth1943** [comment]
Added a screenshot :-)

**Review by nadonomy — CHANGES_REQUESTED** [review]
@resynth1943 this looks great, thanks! 

The body copy is a little verbose at the moment, please can you update to this to merge:

> Your access token gives full access to your account. Do not share it with anyone.

File: src/components/views/dialogs/AccessTokenDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+export default function AccessTokenDialog(props: IProps) {
+    return (
+        <QuestionDialog
+            {...props}
+            title="Reveal Access Token"
+            danger={true}
+            description={_t(
+                "Do not reveal your Access Token to anyone, under any circumstances. " +
+                "Sharing your Access Token with someone would allow them to login to " +
+                "your account, and access your private information.",
```
Comment (resynth1943) [inline src/components/views/dialogs/AccessTokenDialog.tsx:36]:
```suggestion
                "Your access token gives full access to your account. Do not share it with anyone."
```

**Comment by aaronraimist** [comment]
You also need to change it in the json file

File: src/i18n/strings/en_EN.json [hunk 4]
```diff
@@ -1585,6 +1585,7 @@
     "Add a new server...": "Add a new server...",
     "%(networkName)s rooms": "%(networkName)s rooms",
     "Matrix rooms": "Matrix rooms",
+    "Do not reveal your Access Token to anyone, under any circumstances. Sharing your Access Token with someone would allow them to login to your account, and access your private information.": "Do not reveal your Access Token to anyone, under any circumstances. Sharing your Access Token with someone would allow them to login to your account, and access your private information.",
```
Comment (resynth1943) [inline src/i18n/strings/en_EN.json:1588]:
```suggestion
    "Your access token gives full access to your account. Do not share it with anyone.",
```

File: src/i18n/strings/en_EN.json [hunk 5]
```diff
@@ -1585,6 +1585,7 @@
     "Add a new server...": "Add a new server...",
     "%(networkName)s rooms": "%(networkName)s rooms",
     "Matrix rooms": "Matrix rooms",
+    "Your access token gives full access to your account. Do not share it with anyone.",
```
Comment (resynth1943) [inline src/i18n/strings/en_EN.json:1588]:
```suggestion
    "Your access token gives full access to your account. Do not share it with anyone.": "Your access token gives full access to your account. Do not share it with anyone.",
```

File: src/components/views/dialogs/AccessTokenDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+>;
+
+export default function AccessTokenDialog(props: IProps) {
+    return (
+        <QuestionDialog
+            {...props}
+            title="Reveal Access Token"
+            danger={true}
+            description={_t(
+                "Your access token gives full access to your account. Do not share it with anyone."
```
Comment (t3chguy) [inline src/components/views/dialogs/AccessTokenDialog.tsx:34]:
```suggestion
                "Your access token gives full access to your account. Do not share it with anyone.",
```

**Comment by resynth1943** [comment]
@nadonomy do you want me to update the screenshot with the new string?

**Comment by jryans** [comment]
> @nadonomy do you want me to update the screenshot with the new string?

I would advise doing that, yes, since it should make the review much easier.

**Comment by jryans** [comment]
I believe the author is no longer engaged, so it makes sense to close this for now.

**Comment by aaronraimist** [comment]
I've picked up this PR in #5755

**Pull request closed without merging**